benchmarks/src/main/scala/zio/stm/TReentrantLockBenchmark.scala

```diff
@@ -0,0 +1,104 @@
+package zio.stm
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.StampedLock
+
+import scala.util.Random
+
+import org.openjdk.jmh.annotations._
+
+import zio.IOBenchmarks._
+import zio.{ Managed, _ }
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Measurement(iterations = 5, timeUnit = TimeUnit.SECONDS, time = 5)
+@Warmup(iterations = 5, timeUnit = TimeUnit.SECONDS, time = 5)
```

Comment:
I'd increase the number of warmup and measurement iterations to 15.

Reply:
@mijicd - I don't see much difference in the results increasing the iterations but it takes really long to run.

Reply:
Yes, due to the large input matrix it will take longer. However, I don't particularly trust benchmarks with so few iterations, even though they can paint some general picture, they're still prone to noise.

Reply:
Although, we might come to the convention that values in annotation are for such "smoke" runs, while the larger values can be provided via the command line.